### ISSUE TITLE
Specifying nulls

### ISSUE THREAD

**Issue by parautenbach** [ogrodnek/csv-serde#15]
Is there a way to indicate a NULL value? The [default](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL) Hive value is `\N` (capitalised). When I upload a CSV file containing that value, I simply get the value `N`. 

According the the Hive DDL it is not possible to specify a SerDe _and_ NULL terminator. Is there perhaps a class property for this SerDe that can be set (similar to how separator, quote and escape characters are set)? I could not find such a property in the source code. 

It is vital that one can distinguish between NULL and empty values and it would be very helpful if this could be added as a feature.

**Comment by ashrowty** [comment]
I agree with @parautenbach . I have this issue currently. I don't really see the burden .. there needs to be some explicit way to specify what is NULL. Don't mind encoding NULLs as '\N' by default. Maybe this can be an option?

**Comment by huozhanfeng** [comment]
I meet this problem also.

**Comment by bchiud** [comment]
+1

**Comment by sciurus** [comment]
This bit me as well. I have an external table in AWS Athena using OpenCSVSerde. It contains an int column that can be null. If I try to query this column, Athena complains `HIVE_BAD_DATA: Error parsing field value '' for field 0: For input string: ""`

I tried setting the the table property `serialization.null.format'=''` but that did not help.

**Comment by prabhacloud** [comment]
+1

**Comment by sriniprash** [comment]
+1

**Comment by maximvl** [comment]
+1

**Comment by lkudyba** [comment]
Hey @ogrodnek - I know it's an old issue but are you by any chance planning to introduce such a change?

**Comment by ogrodnek** [comment]
Hi @lkudyba ,

This serde code was contributed to Hive back in 2014, and is being maintained there.

See: https://cwiki.apache.org/confluence/display/Hive/CSV+Serde and https://issues.apache.org/jira/browse/HIVE-7777
